pcl_ros/src/pcl_ros/filters/project_inliers.cpp

```diff
@@ -130,42 +131,41 @@ pcl_ros::ProjectInliers::unsubscribe()
   TODO : implement use_indices_
   if (use_indices_)
   {*/
-
   sub_input_filter_.unsubscribe();
+  sub_indices_filter_.unsubscribe();
   sub_model_.unsubscribe();
 }
 
 //////////////////////////////////////////////////////////////////////////////////////////////
 void
 pcl_ros::ProjectInliers::input_indices_model_callback(
-  const PointCloud2::ConstPtr & cloud,
+  const PointCloud2::ConstSharedPtr & cloud,
   const PointIndicesConstPtr & indices,
   const ModelCoefficientsConstPtr & model)
 {
-  if (pub_output_.getNumSubscribers() <= 0) {
+  /*
+   No count_subscribers functionality yet in ROS2
+   if (pub_output_->count_subscribers () <= 0)
```

Comment:
Yes there is!

Reply:
`[get_subscription_count](https://docs.ros2.org/foxy/api/rclcpp/classrclcpp_1_1PublisherBase.html#a60c84260b2ff0fb364fad688ed91234f)`